### PR TITLE
add websocket support

### PR DESCRIPTION
**Pull request by brendanjryan**
Extends https://github.com/mozilla-frontend-infra/react-lazylog/pull/53 with a few refactors and documentation patches.

cc @helfi92, @annaloukianova

### CODE REVIEW

File: src/components/LazyLog/README.md [hunk 1]
```diff
@@ -35,3 +35,29 @@ consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
...
+          socket = sock; sock.send('Socket has been opened!')
+        },
+      formatMessage: (e) => {
+          return JSON.parse(e).message;
+      },
+    }}
+  />
+}
+</div>
+```
```
Comment (helfi92) [inline src/components/LazyLog/README.md]: http://localhost:6060 shows:

Screen Shot 2019-11-08 at 1 38 30 PM

Comment (brendanjryan) [reply src/components/LazyLog/README.md]: @helfi92 thank for looking at this!

what version of node are you running it under? I was not able to repro this issue with either `10.x` or `13.x`

Comment (helfi92) [reply src/components/LazyLog/README.md]: Currently using `v12.13.0`

Comment (helfi92) [reply src/components/LazyLog/README.md]: Ahh. seems to be ok on Chrome but throws on FF.

Comment (brendanjryan) [reply src/components/LazyLog/README.md]: ah interesting! let me dig into that!

**Comment by brendanjryan** [comment]
@helfi92 The issue should be fixed now! There was some broken JSON in the example -- which I'm amazed Chrome even allowed to work...

**Comment by marcoalfonso** [comment]
@helfi92 is there any timeline for when this will be added? Would be really nice for websockets to be supported. Don't want to locally clone my own version of lazy log :3

**Comment by helfi92** [comment]
> @helfi92 The issue should be fixed now!

Awesome, thanks @brendanjryan :)

> @helfi92 is there any timeline for when this will be added?

I will add this during the weekend. I hope that's ok.

File: src/components/LazyLog/README.md [hunk 2]
```diff
@@ -35,3 +35,29 @@ consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
...
+Log viewing using a websocket
+
+```jsx harmony
+const url = 'wss://echo.websocket.org';
+let socket = null;
+
+<div style={{ height: 200, width: 902 }}>
+  <button style={{margin: 5}} onClick={() => socket && socket.send(JSON.stringify({ message: '[taskcluster 2018-11-14 21:08:32.452Z] Worker Group: us-east-1' }))}>ping</button>
+  <LazyLog
+    enableSearch
```
Comment (helfi92) [inline src/components/LazyLog/README.md]: Attempting to search seems to throw `TypeError: rawLog is undefined`.

File: src/components/LazyLog/README.md [hunk 3]
```diff
@@ -35,3 +35,29 @@ consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
...
+    url={url}
+    websocket
+    websocketOptions={{
+      onOpen: (e, sock) => {
+        console.log("opened");
```
Comment (helfi92) [inline src/components/LazyLog/README.md]: We can remove the `console.log` statement.

Comment (brendanjryan) [reply src/components/LazyLog/README.md]: 👍

File: src/components/LazyLog/README.md (hunk 2 above)
Comment (brendanjryan) [reply src/components/LazyLog/README.md]: ah good catch -- this should be fixed now!

File: src/components/LazyLog/README.md [hunk 4]
```diff
@@ -35,3 +35,28 @@ consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?
...
+  <LazyLog
+    enableSearch
+    url={url}
+    websocket
+    websocketOptions={{
+      onOpen: (e, sock) => {
+          socket = sock; sock.send(JSON.stringify({message: "Socket has been opened!"}))
+        },
+      formatMessage: (e) => {
+          return `${JSON.parse(e).message}\n`;
```
Comment (helfi92) [inline src/components/LazyLog/README.md]: Having users write `\n` is a bit inconvenient. Is there a way to handle this internally? Perhaps

```jsx
socket.addEventListener('message', e => {
  let msg = formatMessage ? formatMessage(e.data) : e.data;

  if (!msg.endsWith('\n')) {
    msg += '\n';
  }

  emitter.emit('data', msg);
});
```

Comment (brendanjryan) [reply src/components/LazyLog/README.md]: good suggestion! I added something similar to this in my latest commit.

Thank you for all of your patience and thorough reviews!

**Comment by helfi92** [comment]
Released in `react-lazylog@4.5.0` 🎉

**Comment by brendanjryan** [comment]
awesome - thank you!

**Pull request merged**